### PR TITLE
added exclude spec functionality to cli arguments

### PR DESCRIPTION
**Pull request by PippoRaimondi**
## Proposed changes

Adding the capability to exclude spec(s) file(s) from command line, not a very big piece of work but can be handy... I've also tested the change locally (did some exploratory testing as well and it's ok)

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
 
## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by PippoRaimondi** [comment]
@christian-bromann would it be ok to give me some feedback so we could get this PR moving?

File: lib/utils/ConfigParser.js [hunk 1]
```diff
@@ -272,6 +253,40 @@ class ConfigParser {
...
 
+    /**
+     * sets config attribute with file paths from filtering
+     * options from cli argument
+     *
+     * @param  {String} filterObj  list of files in a string from
+     * cli argument
+     * @return {String} filterType  either exclude or spec
+     */
+    setFilePathToFilterOptions (filterObj, filterType) {
```
Comment (christian-bromann) [inline lib/utils/ConfigParser.js]: function head should be this:

```js
setFilePathToFilterOptions (filterObj, exclude=false) {
```

as I don't see what other filtertype could be used in the future here.

File: lib/utils/ConfigParser.js [hunk 2]
```diff
@@ -272,6 +253,40 @@ class ConfigParser {
...
+        if (typeof filterObj === 'string') {
+            const objs = new Set()
+            const objList = filterObj.split(/,/g)
+            const objsList = ConfigParser.getFilePaths(this._config[filterType])
+
+            for (let obj of objList) {
+                if (fs.existsSync(obj) && fs.lstatSync(obj).isFile()) {
+                    objs.add(path.resolve(process.cwd(), obj))
+                } else {
```
Comment (christian-bromann) [inline lib/utils/ConfigParser.js]: let's add a continue and remove the else to save an identation

**Comment by PippoRaimondi** [comment]
@christian-bromann I've made the modifications you've suggested. I am now working on creating a PR for v5. Thanks for your feedback.

**Comment by PippoRaimondi** [comment]
@christian-bromann I've created a PR for v5 (https://github.com/webdriverio/v5/pull/111)

Please let me know what else should be done to get this PR approved/merged.

**Comment by christian-bromann** [comment]
@asos-filipporaimondi could you please make a new PR against v4 branch? Thanks!

**Pull request closed without merging**